micro-rdk/src/common/conn/server.rs

```diff
@@ -169,6 +175,27 @@ where
             app_connector: self.app_connector,
             app_config: self.app_config,
             max_connections: self.max_connections,
+            app_client_tasks: self.app_client_tasks,
+        }
+    }
+    pub fn with_periodic_app_client_task(
```

Comment:
Continuing the builder pattern here is a little forced because there is no type change that needs to be accommodated, but it seemed best to just do as the other methods do.

Reply:
The thing I like less about that is that it opens up the possibility of using the API wrong, because then you could accidentally call `ViamServer::add_periodic_app_client_task` after `serve` had started to no effect. With it on the builder, there's no confusion, because `build` consumes the builder so you can't call `with_periodic_app_client_task` after `build`, and once you have built you have a `ViamServer` and you can't then do anything but call `serve` on it.

Reply:
this is a great point, I agree with leaving it as is